packages/auth/src/AuthProvider.tsx

```diff
@@ -187,18 +186,32 @@ export class AuthProvider extends React.Component<
    * If the user is assigned any of the provided list of roles,
    * the hasRole is considered to be true.
    */
-  hasRole = (role: string | string[]): boolean => {
-    if (
-      typeof role !== 'undefined' &&
-      this.state.currentUser &&
-      this.state.currentUser.roles
-    ) {
-      if (typeof role === 'string') {
-        return this.state.currentUser.roles?.includes(role) || false
+  hasRole = (rolesToCheck: string | string[]): boolean => {
+    if (typeof rolesToCheck !== 'undefined' && this.state.currentUser?.roles) {
```

Comment:
According to the types `rolesToCheck` can never be `undefined`. Is this code for the benefit of JS projects? Maybe add a comment about that if so. Easy for someone to see this code in the future and not think about that and then remove this check...

Also, could we make this an early return to limit the nesting here a bit.

Reply:
This is for the benefit of JS projects, yes. I am trying to change as little as possible, so I keep any extra checks that are there.

On early return:
Note that I've changed the code below in requireAuth, where hasRoles is called. hasRoles is never called when roles isn't defined, so the check is a bit redundant I suppose

Reply:
> so the check is a bit redundant I suppose

Hehe. So remove it? Or keep it and explain why in a comment?

Reply:
I'm hesitant to remove this, because requireAuth and hasRoles are used in so many different contexts that I can't be sure! What would you suggest to explain in the comment that isn't obvious in the code already?

Reply:
Taking a second look at the code, it should be fine to just remove the check, right?
Let's say `rolesToCheck` is `undefined`.
`if (typeof rolesToCheck === 'string')` will be false. And so will `if (Array.isArray(rolesToCheck))`. So it will fall through to `return false` anyway.

That said, if you do want to keep it I'd do something like this
```
    // Because this code is used from JS projects as well, we can't rely on the
    // typechecking of `rolesToCheck` to protect us from `undefined`
    if (typeof rolesToCheck === 'undefined' || !this.state.currentUser?.roles) {
      return false
    }
```

Reply:
Ok removed, I noticed this is just in auth provider, not the src/lib/auth one. Fine by me!